entities/markets/marketCondition.ts

```diff
@@ -0,0 +1,30 @@
+import { Market } from '@/entities/markets/market';
+import { Condition as ConditionalTokenCondition } from '@/queries/conditional-tokens/types';
+
+export class MarketCondition {
+  market: Market;
+  condition: ConditionalTokenCondition;
+
+  constructor(market: Market, condition: ConditionalTokenCondition) {
+    this.market = market;
+    this.condition = condition;
+  }
+
+  canClaim(index: number) {
+    const isResolved = this.condition.resolved;
+    const hasPayoutDenominator = +this.condition.payoutDenominator > 0;
+
+    return this.market.isWinner(index) && isResolved && hasPayoutDenominator;
+  }
+
+  alreadyClaimed(index: number, outcomeBalance: BigInt | string) {
+    const zero = typeof outcomeBalance === 'bigint' ? BigInt(0) : 0;
+    const balance = typeof outcomeBalance === 'string' ? +outcomeBalance : outcomeBalance;
+
+    return this.canClaim(index) && balance === zero;
+  }
+
+  canRedeem(index: number, outcomeBalance: any) {
+    return this.canClaim(index) && !this.alreadyClaimed(index, outcomeBalance);
+  }
```

Comment:
We could merge these two functions because they do a similar process. Maybe returning a different non-boolean value:

`canClaim && balance === zero` -> claimed
`canClaim && balance !== zero` -> redeemable
`!canClaim` -> early return

Reply:
Having the 2 functions is handy, because we need to know if already claimed to show some things. 
We could use canRedeem but it would not be as descriptive as the name. 

for eg:

```js
 const resultString = alreadyClaimed
            ? 'You claimed:'
            : canClaim
              ? 'You won:'
              : isResolved
                ? 'You lost:'
                : 'You can win:';

```

Already claimed is very readable. I do not bother repeating a bit in this case.